Minor change in the API? Next item doesn't exist on the dictionary

jcintasr opened the pull request:
The _query_datasets method looks for a next item in the dictionary. However, in the case of searching for "fertilizantes" it didn't appeared. Hence, I added a condition to check if the "next" item exists prior to search it.

jcintasr committed:
API changed, fixed based on headers added

jcintasr committed:
Minor fixed: API changed?, NEXT not in the dictionary

The _query_datasets method looks for a next item in the
dicitonary. However, in the case of searching for "fertilizantes" it
didn't appeared. Hence, I added a condition to check if the "next"
item exists prior to search it.

jvaleroliet commented:
Thanks for noting that, I see it cool. If you have time to add the first commit as an option, I will merge.
Thank you!

jcintasr committed:
Added the headers as an optional param to pass

The idea is that, when calling the Manager and OpenDataSet classes,
you can define or not the headers parameter:
- If nothing is defined, then it takes the default value (None)
- If headers is defined as True, then it takes the values porposed in
the stackoverflow answer
https://stackoverflow.com/questions/41946166/requests-get-returns-403-while-the-same-url-works-in-browser
- Also, the user can defined a custom dictionary.

Any other type passed to the parameter will result in a error
indicating the right values to pass.

jcintasr commented:
I think it is done? Can you see the latest commit? Or should I add another pull requests?

Besides, there're more problems with the results the library is returning, but I think they are in the side of the API. For example, if I search for fertilizantes (yes, I'm researching a bit about that), the browser of datos.gob.es returns results that doesn't match with what the API returns (even in their web). So, don't know how to solve that. Maybe I will create an issue or try to contact with them.

Anyway, amazing job :)

jvaleroliet commented:
Congrats for your contribution and thank you so much. Hope it will be helpful and don't hesitate to contribute.

jvaleroliet commented:
I need to upload it to pip too btw

jvaleroliet commented:
Done!

jcintasr commented:
Nice! 😁
